### DISCUSSION TITLE
Question: for desktop app, will the main() be executed multiple times during the app's life time?

### DISCUSSION THREAD

**Discussion by Thenorthernsea** [flet-dev/flet#568] [Q&A]
During the computer's sleep or wake process I noticed sometimes the connection between client and server will lost and the main() is executed during reconnecting

**Comment by Thenorthernsea** [comment]
My app always reconnects after waking from hibernation

**Comment by ndonkoHenri** [comment]
https://github.com/flet-dev/flet/issues/498

https://github.com/flet-dev/flet/issues/546

**Reply by Thenorthernsea** [reply]
Is it normal that main function have to be executed after reconnect and the layout need to be reconstructed?

**Reply by FeodorFitsner** [reply]
`main` function should not be executed on re-connect. Most probably what you observe is a start of a new session. `main` is called on a start of a new session. By default, session is expired in 60 minutes after the last reconnection, so maybe that's the case.

**Reply by Thenorthernsea** [reply]
Hi @FeodorFitsner , is it possible to make the session never expire? I'm building it as a desktop application.

**Reply by ndonkoHenri** [reply]
> is it possible to make the session never expire?

Some more questions:

- How can we change that default expiration time? 
- And also, why is it important to set one..?

**Reply by FeodorFitsner** [reply]
Well, if computer goes to a sleep then it's possible it needs to re-connect to Fletd server on wake up.

Session timeout can be changed via `FLET_APP_LIFETIME_MINUTES` environment variable. The value is minutes.

Try adding this in the beginning of your Python Flet program:

```python
import os
os.environ["FLET_APP_LIFETIME_MINUTES"] = "1440" # 24 hours
```
